src/hotspot/share/oops/instanceKlass.hpp

```diff
@@ -715,16 +715,16 @@ class InstanceKlass: public Klass {
   }
 
  private:
-  static bool  _has_previous_versions;
+  static bool  _clean_previous_versions;
```

Comment:
Nit: I'd suggest to name it as `_should_clean_previous_versions`.
       Then the corresponding function needs to be named as `should_clean_previous_versions()`.